### PR TITLE
Gorgoris/dao/invoice

### PR DESCRIPTION
**Pull request by metagon**

### CODE REVIEW

File: src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBInvoiceDAO.java [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    public void create(Invoice invoice) throws DAOException, ValidationException {
+        LOGGER.debug("Entering create with parameters: " + invoice);
+
+        invoiceValidator.validateForCreate(invoice);
+
+        final String query = "INSERT INTO Invoice (invoiceTime, brutto, user_ID) " +
+                "VALUES (?, ?, ?);";
+        try (PreparedStatement stmt = dataSource.getConnection()
+                .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            stmt.setTimestamp(1, invoice.getTime() == null ? null : Timestamp.valueOf(invoice.getTime()));
```
Comment (emmanuel099) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBInvoiceDAO.java]: can not be null

File: src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBInvoiceDAO.java [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+    @Override
+    public void update(Invoice invoice) throws DAOException, ValidationException {
+        LOGGER.debug("Entering update with parameters: " + invoice);
+
+        invoiceValidator.validateForUpdate(invoice);
+
+        final String query = "UPDATE Invoice SET invoiceTime = ?, brutto = ?, user_ID = ? " +
+                "WHERE id = ?;";
+        try (PreparedStatement stmt = dataSource.getConnection().prepareStatement(query)) {
+            stmt.setTimestamp(1, invoice.getTime() == null ? null : Timestamp.valueOf(invoice.getTime()));
```
Comment (emmanuel099) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBInvoiceDAO.java]: can't be null

File: src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBInvoiceDAO.java [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+        final String query = "SELECT * FROM Invoice WHERE " +
+                "ID = ISNULL(?, ID) AND " +
+                "invoiceTime = ISNULL(?, invoiceTime) AND " +
+                "brutto = ISNULL(?, brutto) AND " +
+                "user_ID = ISNULL(?, user_ID) AND " +
+                "canceled = FALSE;";
+        try (PreparedStatement stmt = dataSource.getConnection().prepareStatement(query)) {
+            stmt.setLong(1, invoice.getIdentity());
+            stmt.setTimestamp(2, invoice.getTime() == null ? null : Timestamp.valueOf(invoice.getTime()));
```
Comment (emmanuel099) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBInvoiceDAO.java]: setObject and remove == null check

Comment (metagon) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBInvoiceDAO.java]: setObject is not working. Tried it and it failed

**Pull request merged**